rollout/trafficrouting/istio/istio.go

```diff
@@ -209,17 +209,26 @@ func (r *Reconciler) reconcileVirtualService(obj *unstructured.Unstructured, des
 		}
 	}
 
+	// Generate Patches
 	patches := r.generateVirtualServicePatches(httpRoutes, tlsRoutes, int64(desiredWeight))
 	err = patches.patchVirtualService(httpRoutesI, tlsRoutesI)
 	if err != nil {
 		return nil, false, err
 	}
 
-	err = unstructured.SetNestedSlice(newObj.Object, httpRoutesI, "spec", Http)
-	if err != nil {
-		return newObj, len(patches) > 0, err
+	// Set HTTP Route Slice
+	if len(httpRoutes) > 0 {
+		err = unstructured.SetNestedSlice(newObj.Object, httpRoutesI, "spec", Http)
+		if err != nil {
+			return newObj, len(patches) > 0, err
+		}
 	}
-	err = unstructured.SetNestedSlice(newObj.Object, tlsRoutesI, "spec", Tls)
```

Comment:
Any chance we can have unit test to cover the above lines; else looks good.

Reply:
No, it's just a single line change so I haven't really added any UTs.